src/python_testing/TC_IDM_1_4.py

```diff
@@ -247,8 +247,65 @@ async def steps_3_to_9(self, dummy_value):
         except InteractionModelError:
             asserts.fail("DUT failed with non-path specific error when path specific error was expected")
 
-        # Skipping test until https://github.com/project-chip/connectedhomeip/issues/31434 resolved
-        self.skip_step(9)
+        self.step(9)
+        try:
+            feature_map = await self.read_single_attribute(
+                dev_ctrl,
+                dut_node_id,
+                endpoint=0,
+                attribute=Clusters.GeneralDiagnostics.Attributes.FeatureMap
+            )
+        except InteractionModelError:
+            asserts.fail("DUT failed to respond reading FeatureMap attribute")
+        has_data_model_test_feature = (feature_map & Clusters.GeneralDiagnostics.Bitmaps.Feature.kDataModelTest) != 0
+        asserts.assert_true(has_data_model_test_feature, "DataModelTest Feature is not supported by DUT")
+
+        self.step(10)
+        try:
+            test_event_triggers_enabled = await self.read_single_attribute(
+                dev_ctrl,
+                dut_node_id,
+                endpoint=0,
+                attribute=Clusters.GeneralDiagnostics.Attributes.TestEventTriggersEnabled
+            )
+        except InteractionModelError:
+            asserts.fail("DUT failed to respond reading TestEventTriggersEnabled attribute")
+        asserts.assert_true(test_event_triggers_enabled, "Test Event Triggers are NOT Enabled on DUT")
+
+        self.step(11)
+        asserts.assert_true('PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY' in self.matter_test_config.global_test_params,
+                            "PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY must be included on the command line in "
+                            "the --hex-arg flag as PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY:<key>, "
+                            "e.g. --hex-arg PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY:000102030405060708090a0b0c0d0e0f")
+        enable_key = self.matter_test_config.global_test_params['PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY']
+        endpoint = 0
+        command = Clusters.GeneralDiagnostics.Commands.PayloadTestRequest(
+            enableKey=enable_key,
+            value=ord('A'),
+            count=800
+        )
+        invoke_request_1 = Clusters.Command.InvokeRequestInfo(endpoint, command)
+
+        command = Clusters.OperationalCredentials.Commands.CertificateChainRequest(
+            Clusters.OperationalCredentials.Enums.CertificateChainTypeEnum.kDACCertificate)
+        invoke_request_2 = Clusters.Command.InvokeRequestInfo(endpoint, command)
+
+        try:
+            result = await dev_ctrl.TestOnlySendBatchCommands(dut_node_id, [invoke_request_1, invoke_request_2])
+        except InteractionModelError:
+            asserts.fail("DUT failed to respond to")
+
+        responses = result.Responses
```

Comment:
The `TestOnlySendBatchCommands` should return a list, rather than an object with a .Responses list arg. It would be much more discoverable. Can be done as follow-up.

Reply:
I disagree and you approved the change where we moved this from a list of response to a structure with extra data so we can actually get the InvokeResposeMessage Count